R/animint.R

```diff
@@ -649,12 +694,26 @@ saveLayer <- function(l, d, meta){
 saveChunks <- function(x, vars, meta){
   if(is.data.frame(x)){
     if(length(vars) == 0){
+      if(meta$geom %in% c("line", "path", "polygon", "ribbon")) {
+        if("group" %in% names(x)) {
+          # common info for groups, save only once
+          csv.name <- sprintf("%s_chunk_shape.tsv", meta$classed)
+          chunk.shape <- file.path(meta$out.dir, csv.name)
+          if(!file.exists(chunk.shape)){
+            x1 <- x[c("x", "y", "group")]
+            write.table(x1, chunk.shape, quote = FALSE, row.names = FALSE, 
+                        sep = "\t")
+          }
+          # info for each group
+          x2 <- x[, !names(x) %in% c("x", "y")]
+          x <- x2[!duplicated(x2), ]
+          }
```

Comment:
I think the redundancy mostly occurs when drawing spatial objects (excluding point). For non-spatial objects, it may occur the join relation of n:1 for constant fields and updating fields. If I split variable `x` based on whether repeated data present, how do I deal with the case that the updating field indeed has duplicated values, for example in my `geom_polygon` example, some states are of the same fill colors.

Besides, I am not clear that what should I test if write a test for this function? Could you give a hint?